pynest/examples/hpc_benchmark.py

```diff
@@ -0,0 +1,487 @@
+   In order to make it possible to test this benchmark script on desktop
+   computers, the scale parameter is set to 1 below, while the number of
+   11.250 incoming synapses per neuron is retained. In this limit, correlations
+   in input to neurons are large and will lead to increasing synaptic weights.
+   Over time, network dynamics will therefore become unstable and all neurons
+   in the network will fire in synchrony, leading to extremely slow simulation
+   speeds.
+
+   Therefore, the presimulation time is reduced to 50 ms below and the
+   simulation time to 250 ms, while we usually use 100 ms presimulation and
+   1000 ms simulation time.
+
+   For meaningful use of this benchmark, you should use a scale > 10 and check
+   that the firing rate reported at the end of the benchmark is below 10 spikes
+   per second.
+'''
+
+import numpy as np
+import os
+import sys
+import time
+
+import nest
+import nest.raster_plot
+
+M_INFO = 10
+M_ERROR = 30
+
+
+'''Parameter section
+
+ Define all relevant parameters: changes should be made here
+
+'''
+
+params = {
+    'nvp': 1,               # total number of virtual processes
+    'scale': 1.,           # scaling factor of the network size,
+                            # total network size = scale*11250 neurons
+    'simtime': 250.,        # total simulation time in ms
+    'presimtime': 50.,      # simulation time until reaching equilibrium
+    'dt': 0.1,              # simulation step
+    'record_spikes': True,  # switch to record spikes of excitatory
+                            # neurons to file
+    'path_name': '',        # path where all files will have to be written
+    'log_file': 'log',      # naming scheme for the log files
+}
+
+# -----------------------------------------------------------------------------
+
+
+def convert_synapse_weight(tau_m, tau_syn, C_m):
+    '''Computes conversion factor for synapse weight from mV to pA
+
+    This function is specific to the leaky integrate-and-fire neuron
+    model with alpha-shaped postsynaptic currents.
+
+    '''
+    a = tau_m / tau_syn
+    b = 1.0 / tau_syn - 1.0 / tau_m
+    # time of maximum
+    t_max = 1.0 / b * (-lambertwm1(-np.exp(-1.0 / a) / a).real - 1.0 / a)
+    # maximum of PSP for current of unit amplitude
+    v_max = np.exp(1.0) / (tau_syn * C_m * b) * (
+        (np.exp(-t_max / tau_m) - np.exp(-t_max / tau_syn)) /
+        b - t_max * np.exp(-t_max / tau_syn))
+    return 1. / v_max
+
+
+# Rise time of synaptic currents
+# The synaptic rise time is chosen such that the rise time of the
+# evoked post-synaptic potential is 1.700759 ms.
+# For alpha-shaped postynaptic currents, the rise time of the post-synaptic
+# potential follows from the synaptic rise time as
+#
+# def PSP_rise_time(tau_m, tau_syn):
+#     a = tau_m / tau_syn
+#     b = 1.0 / tau_syn - 1.0 / tau_m
+#
+# Inverting this equation numerically leads to tau_syn =
+# 0.32582722403722841 ms, as specified in model_params below.
```

Comment:
@mschmidt87 I am a bit confused here, and since you inserted the corresponding comments/code in the SLI version of the benchmark a while ago, I hope you can clarify what is happening here. My first concern is that in the function `PSP_rise_time()` above a line seems to be missing after the two lines defining `a` and `b`---maybe the line defining `t_max` on line 111 above? As it is written now, there is nothing really to invert.

The other question is why we want to achieve the rise time of 1.700759 ms?

Reply:
@mschmidt87 I am missing a line in the sli and the py version of hpc_benchmark. The `PSP_rise_time` function, as far as I can see, only sets local variables `a` and `b`, but does not compute any rise time and does not return any value. I *assume* that the rise time the function is supposed to compute is the `t_max` expression in function `ConvertSynapseWeight`. Probably someone here planned to split that function up, forgot to copy a line, and then never followed quite through. If you agree with this interpretation, we could change both versions consistently.
The duplicate definition of `/tau_syn` looks like a code duplication introduced by a merge.

Reply:
Yes, you are absolutely right, I agree to your interpretation, sorry for being a bit slow-witted here. Line 111 should be moved inside `PSP_rise_time`, the code then be commented in, and lines 108 and 111 can then be replaced by a call of `PSP_rise_time` (`b` is still needed in the rest of the function). Also, `t_max` can be renamed to `t_rise`, or at least we should settle on either "time of maximum" or "rise time".